### PR TITLE
Enforce idiomatic error handling in bash

### PR DESCRIPTION
**Pull request by senekor**
Single commands that may error can stil be achieved with `|| true`.
With this change, we'll only see a limited number of clippy warnings
in CI. I think that's fine, the one fixing it can still remove these
flags locally while working on the fix to see all warnings.

### CODE REVIEW

**Comment by senekor** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1736** <a href="https://app.graphite.dev/github/pr/exercism/rust/1736" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/exercism/rust/1736?utm_source=stack-comment).

**Comment by senekor** [comment]
I became aware that this is an issue when I saw that CI passed on another branch, even though it shouldn't have:
https://github.com/exercism/rust/actions/runs/6175508949/job/16762524913?pr=1723

This first commit reveals the problem.

File: bin/check_exercises.sh [hunk 1]
```diff
@@ -33,39 +28,39 @@ return_code=0
 # An exercise worth testing is defined here as any top level directory with
 # a 'tests' directory and a .meta/config.json.
 for exercise in $files; do
-   exercise="$exercise/$target_dir"
+    exercise="$exercise/$target_dir"
```
Comment (ErikSchierboom) [inline bin/check_exercises.sh:31]: The formatting changes seem to be unrelated to the thing the PR is fixing. I'm happy to approve, but would recommend keeping unrelated in separate PRs in the future. It makes reviewing easier.

**Review by ErikSchierboom — APPROVED** [review]

File: bin/check_exercises.sh (hunk 1 above)
Comment (senekor) [reply bin/check_exercises.sh:31]: alright, will do!

**Pull request merged**